public/app/features/provisioning/api/types.ts

```diff
@@ -24,13 +26,14 @@ export type S3RepositoryConfig = {
 export type RepositorySpec = {
   title?: string;
   description?: string;
-  folder?: string;
+  sync: SyncOptions;
+  editing: EditingOptions;
+
+  type: 'github' | 'local' | 's3';
+
   github?: GitHubRepositoryConfig;
```

Comment:
These types shouldn't really be used, there are the same types in `endpoints.gen` file, which automatically kept updated.

Reply:
ya -- i was suspicious when editing; but the alternative seems to require a much bigger refactor

Reply:
These type changes look identical to the generated ones to me.

Reply:
I was also not too convinced when having to touch those files :) good to know now :)